src/Microsoft.AspNet.Mvc.TagHelpers/TagHelperOutputExtensions.cs

```diff
@@ -100,5 +110,61 @@ public static class TagHelperOutputExtensions
                 tagHelperOutput.Attributes.Remove(attribute);
             }
         }
+
+        private static void CopyHtmlAttribute(
+            int allAttributeIndex,
+            TagHelperOutput tagHelperOutput,
+            TagHelperContext context)
+        {
+            var existingAttribute = context.AllAttributes[allAttributeIndex];
+            var copiedAttribute = new TagHelperAttribute
+            {
+                Name = existingAttribute.Name,
+                Value = existingAttribute.Value,
+                Minimized = existingAttribute.Minimized
+            };
+
+            // Move backwards through context.AllAttributes from the provided index until we find a familiar attribute
+            // in tagHelperOutput where we can insert the copied value in front of.
+            for (var i = allAttributeIndex - 1; i >= 0; i--)
+            {
+                var previouName = context.AllAttributes[i].Name;
+                var index = IndexOfFirstMatch(previouName, tagHelperOutput.Attributes);
```

Comment:
Suspect this code needs to figure out how many instances of `previousName` are in `context.AllAttributes[0..allAttributeIndex -1]` and look forward through `output.Attributes` to find the same count (or the last match).